internal/cba.go

```diff
@@ -275,8 +265,25 @@ func GetHTTPTransportConfigAndEndpoint(settings *DialSettings) (cert.Source, fun
 	return nil, dialTLSContextFunc, config.s2aMTLSEndpoint, nil
 }
 
+func shouldUseS2A(clientCertSource cert.Source, settings *DialSettings) bool {
+	// If client cert is found, use that over S2A.
+	// If EXPERIMENTAL_GOOGLE_API_USE_S2A is not set to true, skip S2A.
+	// If DefaultMTLSEndpoint is not set, skip S2A.
+	// If MTLS is not enabled for this endpoint, skip S2A.
+	// If custom HTTP client is provided, skip S2A.
+	if clientCertSource != nil || !isGoogleS2AEnabled() || settings.DefaultMTLSEndpoint == "" ||
+		!mtlsEndpointEnabledForS2A() || settings.HTTPClient != nil {
+		return false
```

Comment:
nit: I would consider unrolling this to multiple conditionals to improve the readability. Given there are 5 boolean expressions, and two of them with a negation, it is hard to follow the logic.

Reply:
done thanks for the comments!